x-pack/legacy/plugins/monitoring/public/views/no_data/controller.js

```diff
@@ -12,32 +12,76 @@ import {
   startChecks
 } from 'plugins/monitoring/lib/elasticsearch_settings';
 import { ModelUpdater } from './model_updater';
-import { render, unmountComponentAtNode } from 'react-dom';
 import { NoData } from 'plugins/monitoring/components';
 import { timefilter } from 'ui/timefilter';
 import { I18nContext } from 'ui/i18n';
 import { CODE_PATH_LICENSE } from '../../../common/constants';
+import { MonitoringViewBaseController } from '../base_controller';
+import { i18n } from '@kbn/i18n';
 
-const REACT_NODE_ID_NO_DATA = 'noDataReact';
+export class NoDataController extends MonitoringViewBaseController {
 
-// NoDataController watches all this attributes.
-// After consulting with @Chris Roberson, decided to move this out of the class for now.
-let timeUpdateSubscription;
+  constructor($injector, $scope) {
+    const monitoringClusters = $injector.get('monitoringClusters');
+    const kbnUrl = $injector.get('kbnUrl');
+    const $http = $injector.get('$http');
+    const checkers = [
+      new ClusterSettingsChecker($http),
+      new NodeSettingsChecker($http)
+    ];
 
-export class NoDataController {
+    let watcherSet;
 
-  constructor($injector, $scope) {
-    const $executor = $injector.get('$executor');
-    this.enableTimefilter($executor);
-    this.registerCleanup($scope, $executor);
+    const getData = async () => {
+      let catchReason;
+      try {
+        const monitoringClustersData = await monitoringClusters(undefined, undefined, [CODE_PATH_LICENSE]);
+        if (monitoringClustersData && monitoringClustersData.length) {
+          kbnUrl.redirect('/home');
+          return monitoringClustersData;
+        }
+      }
+      catch (err) {
+        if (err && err.status === 503) {
+          catchReason = {
+            property: 'custom',
+            message: err.data.message,
+          };
+        }
+      }
+
+      const { updateModel } = new ModelUpdater($scope, this);
+      const enabler = new Enabler($http, updateModel);
+      if (!watcherSet) {
+        watcherSet = $scope.$watch(() => this, () => this.render(enabler), true);
+      }
 
+      if (catchReason) {
+        this.reason = catchReason;
+      } else if (!this.isCollectionEnabledUpdating
+        && !this.isCollectionEnabledUpdated
```

Comment:
So I was working through the process of enabling monitoring while having some misconfigurations in my cluster settings. For example, I had monitoring disabled, but I also had a local exporter configured that was also disabled (a local exporter disabled will show an error on this page). When I clicked the blue button to enable monitoring, the spinner never left, but after some debugging I realized it was because `this.isCollectionEnabledUpdated` was set to true and never reset. 

I don't know if the right change is to remove this line from the conditional, or maybe look into how it's mutated to understand if a change is necessary there, but I think we can do a little more to make a more seamless UX.

Reply:
Looks like it's only set in the catch but never reseted: https://github.com/elastic/kibana/blob/7ac8e4d9ccc2a9e064d8a5621cd62453d1006164/x-pack/legacy/plugins/monitoring/public/lib/elasticsearch_settings/enabler.js#L47

I will set it to false before the async function here: https://github.com/elastic/kibana/blob/7ac8e4d9ccc2a9e064d8a5621cd62453d1006164/x-pack/legacy/plugins/monitoring/public/lib/elasticsearch_settings/enabler.js#L38

Reply:
I don't think this solution quite works. I'm not seeing a change in behavior on my side. It's still stuck on `true` which is preventing the next `startChecks()` call

Reply:
Maybe we're testing different scenarios. What steps are taking to reproduce the sticky loading phase?

Reply:
Sure, so start off with a cluster that has no monitoring data and does not have monitoring enabled. Go to the dev tools and configure a local or remote exporter that is disabled, something like:
```
PUT _cluster/settings
{
  "transient": {
    "xpack.monitoring.exporters": { 
      "my_remote": {
        "type": "http",
        "enabled": false
      }
    }
  }
}
```

Go to the monitoring UI, see the blue enable button, and click it. My experience is it just keeps spinning and never shows me the next error (which is that all my exporters are disabled).

This is where I found the list of scenarios that would cause an error message: https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/monitoring/server/lib/elasticsearch_settings/find_reason.js

Reply:
Actually I think the best solution would be to remove that if condition, and just let the checks happen

@chrisronline What do you think?

Reply:
Sounds good to me as long as it solves this problem and doesn't introduce any additional issues, 👍